eahub/base/tests/test_base_models.py

```diff
@@ -0,0 +1,15 @@
+from django.test import TestCase
+from model_mommy import mommy
+
+
+class UserTests(TestCase):
+    def test_has_profile(self):
+        user = mommy.make("base.User")
+
+        # With no profile
+        self.assertFalse(user.has_profile())
+
+        # With a profile
+        mommy.make("Profile", user=user, slug="slug")
```

Comment:
I worry that this isn't a good unit test because there's a dependency on another class (Profile) here. But I don't know of another way to test the behaviour of the method under test here, than to use a real Profile object. In other words, I'm not sure if in this case the most isolated type of test possible is an integration test with such a dependency.

Reply:
(Question still applies, but commit has moved after a git rebase.)

Reply:
This is a common misconception; it's totally fine to use real dependencies in unit tests, unless there's a specific reason why doing so inhibits hermeticity or is otherwise problematic. (There's some really good documentation about this at my workplace that I'm trying to get open sourced.)

Reply:
Perhaps you can point me to some docs that exist outside your company @taymonbeal ?

Reply:
I'll try to find some. In the meantime, the really short version is, you don't have to swap out dependencies of the code under test; it's fine to use the real dependencies unless there's a specific technical reason why this is impossible or undesirable.